hw6-2/cycleList.c

```diff
@@ -0,0 +1,196 @@
+#include <stdlib.h>
+#include <stdbool.h>
+#include <stdio.h>
+
+typedef struct CycleListElement {
+    int value;
+    struct CycleListElement *next;
+} CycleListElement;
+
+typedef struct CycleList {
+    CycleListElement *firstvalue;
+} CycleList;
+
+CycleList *createList() {
+    return calloc(1, sizeof(CycleList));
+}
+
+int push(CycleList *cycleList, int value) {

```

Comment:
```suggestion
int add(CycleList *cycleList, int value) {
```
push - стандартное имя для стека, в случае списков это add.

Reply:
А эта функция - addToEnd, так как добавляем в конец.
Такая вставка имеет линейную сложность. Поэтому, если мы вставляем N элементов, мы получим сложность O(n^2). А можно за линию, если вставлять в начало, а в структуре хранить ссылку на последний элемент.